omnigibson/action_primitives/action_primitive_set_base.py

```diff
@@ -0,0 +1,76 @@
+import inspect
+from abc import ABCMeta, abstractmethod
+from enum import IntEnum
+from typing import List
+
+from future.utils import with_metaclass
+
+from omnigibson.robots import BaseRobot
+from omnigibson.scenes.interactive_traversable_scene import InteractiveTraversableScene
+from omnigibson.tasks.task_base import BaseTask
+
+REGISTERED_PRIMITIVE_SETS = {}
+
+class ActionPrimitiveError(ValueError):
+    class Reason(IntEnum):
+        # The planning for a primitive was successfully completed, but an error occurred during execution.
+        EXECUTION_ERROR = 0
```

Comment:
nit: can we order them so the numbers increase as the skill is executed chronologically?

i.e.: precondition --> sampling --> planning --> execution --> postcondition

Reply:
Done.